src/components/lab/ElectricFieldVectorGroups.js

```diff
@@ -4,72 +4,50 @@ import { Group } from 'react-konva';
 import PropTypes from 'prop-types';
 import ElectricFieldVectorGroup from './ElectricFieldVectorGroup';
 import {
+  GREENHOUSE_GASES,
   INFRARED_RADIATION_CURVE_INTERVAL,
-  INFRARED_RADIATION_PERIOD,
-  INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL,
-  SINE_CURVE_AMPLITUDE,
+  MOLECULE_CENTER_Y_FROM_BOTTOM_OF_CANVAS,
   Y_SHIFT_PER_INTERVAL,
 } from '../../config/constants';
+import generateElectricFieldVectorGroupYPoints from '../../utils/generateElectricFieldVectorGroupYPoints';
 
-const ElectricFieldVectorGroups = ({ x }) => {
+const ElectricFieldVectorGroups = ({ x, groupIndex }) => {
   const { height: stageHeight } = useSelector(
     ({ layout }) => layout.lab.stageDimensions,
   );
   const intervalCount = useSelector(({ lab }) => lab.intervalCount);
+  const moleculesOnCanvas = useSelector(({ lab }) => lab.moleculesOnCanvas);
+  const currentLineMolecule = moleculesOnCanvas[groupIndex].molecule;
+
+  let absorptionPoint = 0;
+  if (GREENHOUSE_GASES.includes(currentLineMolecule)) {
+    absorptionPoint = stageHeight - MOLECULE_CENTER_Y_FROM_BOTTOM_OF_CANVAS;
+  }
 
   return (
     <Group>
-      <ElectricFieldVectorGroup
-        x={
-          x -
-          SINE_CURVE_AMPLITUDE +
-          SINE_CURVE_AMPLITUDE *
-            Math.sin(
-              (intervalCount -
-                INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL) *
-                Y_SHIFT_PER_INTERVAL *
-                INFRARED_RADIATION_PERIOD,
-            )
-        }
-        y={stageHeight - INFRARED_RADIATION_CURVE_INTERVAL}
-        direction={1}
-      />
-      <ElectricFieldVectorGroup
-        x={
-          x +
-          SINE_CURVE_AMPLITUDE -
-          SINE_CURVE_AMPLITUDE *
-            Math.sin(
-              (intervalCount -
-                INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL) *
-                Y_SHIFT_PER_INTERVAL *
-                INFRARED_RADIATION_PERIOD,
-            )
-        }
-        y={stageHeight - 3 * INFRARED_RADIATION_CURVE_INTERVAL}
-        direction={-1}
-      />
-      <ElectricFieldVectorGroup
-        x={
-          x -
-          SINE_CURVE_AMPLITUDE +
-          SINE_CURVE_AMPLITUDE *
-            Math.sin(
-              (intervalCount -
-                INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL) *
-                Y_SHIFT_PER_INTERVAL *
-                INFRARED_RADIATION_PERIOD,
-            )
-        }
-        y={stageHeight - 5 * INFRARED_RADIATION_CURVE_INTERVAL}
-        direction={1}
-      />
+      {generateElectricFieldVectorGroupYPoints(
+        intervalCount,
+        stageHeight,
+        absorptionPoint,
+        INFRARED_RADIATION_CURVE_INTERVAL,
+        Y_SHIFT_PER_INTERVAL,
+      ).map(({ y, direction }, index) => (
+        <ElectricFieldVectorGroup
+          x={x}
+          y={y}
+          direction={direction}
+          // eslint-disable-next-line react/no-array-index-key
+          key={index}
```

Comment:
Would  `y`  work as a key?

Reply:
ever since #15 I have been reluctant to use canvas coordinates as keys for components. using `index` breaks the eslint rule but guarantees that the keys are unique and won't be randomly recalculated, causing components to unmount...